lib/IRGen/GenDecl.cpp

```diff
@@ -1648,7 +1648,7 @@ getIRLinkage(const UniversalLinkageInfo &info, SILLinkage linkage,
     if (isDefinition) {
       return std::make_tuple(llvm::GlobalValue::AvailableExternallyLinkage,
                              llvm::GlobalValue::DefaultVisibility,
-                             ExportedStorage);
+                             llvm::GlobalValue::DefaultStorageClass);
```

Comment:
Wow, this is a bit tricky to reason about.  Here, you have a symbol marked as `available_externally` (which means that it is available for inspection, but not emission).  As a result, it should never be exported *nor* imported.  If the definition is not meant to be used, should it not be assumed that it will be imported from another module?

Reply:
I agree in principle that this setting should not matter because the value should never be emitted. However, because dllexport/dllimport involve adding separate "__imp_" symbols, it seems better to use the default here. I'm not aware of anything breaking because of this one, though.

Reply:
Markdown ate that symbol prefix. It should be "\_\_imp\_"

Reply:
The `__imp_` prefix is added very late in the backend, and getting that right ensures that we get better code if we are indirecting through the pointer for a function call.  That said, since the value will never be emitted, we can deal with the optimization scenario later.